ben-gb/server.js

```diff
@@ -0,0 +1,189 @@
+  let values = [
+    request.body.title,
+    request.body.author,
+    request.body.authorUrl,
+    request.body.category,
+    request.body.publishedOn,
+    request.body.body,
+    request.params.id
+  ];
+
+  client.query( SQL, values )
+    .then(() => {
+      response.send('update complete')
+    })
+    .catch(err => {
+      console.error(err);
+    });
+});
+
+app.delete('/articles/:id', (request, response) => {
+  // COMMENT: What number(s) of the full-stack-diagram.png image correspond to the following line of code? Which method of article.js is interacting with this particular piece of `server.js`? What part of CRUD is being enacted/managed by this particular piece of code?
+  // This is associated with step 3, it is the server telling the database to delete a specified row of information from the table. It is associated with the deleteRecord method in article JS. It is a D in the CRUD process in that it is deleting a record.
+
+  let SQL = `DELETE FROM articles WHERE article_id=$1;`;
+  let values = [request.params.id];
+
+  client.query( SQL, values )
+    .then(() => {
+      response.send('Delete complete')
+    })
+    .catch(err => {
+      console.error(err);
+    });
+});
+
+app.delete('/articles', (request, response) => {
+  // COMMENT: What number(s) of the full-stack-diagram.png image correspond to the following line of code? Which method of article.js is interacting with this particular piece of `server.js`? What part of CRUD is being enacted/managed by this particular piece of code?
+  // this is associated with step three, it is deleting all of our records in the SQL database upon execution. The server tells the database upon the call of truncateTable in article.js to delete all records. It is a D in the crud process.
+
+  let SQL = `DELETE FROM articles;`;
+  client.query( SQL )
+    .then(() => {
+      response.send('Delete complete')
+    })
+    .catch(err => {
+      console.error(err);
+    });
+});
+
+// COMMENT: What is this function invocation doing?
+// It creates a database if one doesn't exist and loads the database from hackerIpsum if the database is empty.
+loadDB();
+
+app.listen(PORT, () => {
+  console.log(`Server started on port ${PORT}!`);
+});
+
+
+//////// ** DATABASE LOADER ** ////////
+////////////////////////////////////////
+function loadArticles() {
+  // COMMENT: What number(s) of the full-stack-diagram.png image correspond to the following line of code? Which method of article.js is interacting with this particular piece of `server.js`? What part of CRUD is being enacted/managed by this particular piece of code?
+  // This is another step three, that is going from the server to the database and returns information to the server. It is not associated with a method in article.js. It is associated with C in crud as it is querying then creating what is in the SQL database and returning data if it exists.
```

Comment:
This is not particularly one of the numbers in the diagram; it's something that runs once when the server is started.